strictdoc/export/html/generators/view_objects/project_statistics_view_object.py

```diff
@@ -34,19 +36,19 @@ def __init__(
         self.is_running_on_server: bool = project_config.is_running_on_server
         self.strictdoc_version = __version__
 
-    def render_screen(self, jinja_environment: JinjaEnvironment):
+    def render_screen(self, jinja_environment: JinjaEnvironment) -> Markup:
         return jinja_environment.render_template_as_markup(
             "screens/project_statistics/index.jinja", view_object=self
         )
 
-    def render_static_url(self, url: str):
-        return self.link_renderer.render_static_url(url)
+    def render_static_url(self, url: str) -> Markup:
```

Comment:
Would it make sense to move the `Markup(...)` down to the link_renderer? Because that's where that content originates?
This comment applies to all occurrences of this change.

Reply:
I left this comment but let's discuss it independently or this MR which I will merge now as-is.

Reply:
> move the Markup(...) down to the link_renderer

The "problem" is, that `Markup` is "infectious". Any `str` that's somehow mixed (+, replace, ...) with `Markup` will itself become `Markup` and gets escaped on the way. This is not always what we want. Lower level functions may produce output that's not only meant to end up in HTML, but also into other formats. HTML escaping would be inadequate there, but it's hard to control if Markup objects are unexpectedly passed around.

My conclusion until now is that returning `Markup` is good, but only if we can guarantee that a function is only ever used in HTML context.